### ISSUE TITLE
Messages section lists warnings multiple times

### ISSUE THREAD

**Issue by freeman42x** [atom-haskell/ide-haskell-repl#73]
Following is shown after loading REPL in `Utils.hs`:

![image](https://user-images.githubusercontent.com/1758708/79388603-851e4780-7f65-11ea-9ed3-cc27d9965b6e.png)

As you can see the warnings are printed 3 times.
If I open `cabal repl` in that project the warnings are only listed once and the extension is configured to use cabal:

![image](https://user-images.githubusercontent.com/1758708/79388736-c282d500-7f65-11ea-964d-fc5299024a7a.png)

**Comment by lierdakil** [comment]
Hm. I understand where one duplicate might come from, but not the other. If you'd be using ide-haskell-repl with ide-haskell, then you wouldn't see those duplicates anyway, since ide-haskell handles duplicate messages transparently. One duplicate comes from the fact that ide-haskell-repl has to load the particular source file it was invoked on after `cabal repl` starts. This should be possible to filter out, although I'm somewhat reluctant to do so, due to those not necessarily being duplicates. The other one I honestly have no idea at the moment.

**Comment by lierdakil** [comment]
Ah. I guess I see how the second one comes about. That's something GHCi does apparently when `:set +c` is enabled:
```
*ManData> :set +c             
*ManData> :load src/ManData.hs
[1 of 1] Compiling ManData          ( src/ManData.hs, interpreted )

src/ManData.hs:32:1: warning: [-Wmissing-signatures]
    Top-level binding with no type signature: f :: a -> a
   |
32 | f x = id x
   | ^
Ok, one module loaded.
Collecting type info for 1 module(s) ... 

src/ManData.hs:32:1: warning: [-Wmissing-signatures]
    Top-level binding with no type signature: f :: a -> a
   |
32 | f x = id x
   | ^
```

Welp, I guess I have to filter out duplicates the hard way then...

**Comment by freeman42x** [comment]
@lierdakil I am not using `:set +c` in the project `.ghci`: https://github.com/fairy-tale-agi-solutions/haskell-editor-setup/blob/master/.ghci

And I do not have any other non-project level `.ghci` afaik.

**Comment by lierdakil** [comment]
`:set +c` is something ide-haskell-repl does for reasons. see https://github.com/atom-haskell/ide-haskell-repl/blob/67b2d2c78d3193ca1076cbbed656fdacd1ad61bc/src/ghci.ts#L54-L58

**Comment by lierdakil** [comment]
Should be fixed in v0.9.6
